jit-service/src/main/java/org/dcsa/jit/service/TimestampRoutingService.java

```diff
@@ -4,44 +4,160 @@
+      .choice()
+        // outbox_message has a not null constraint on message_routing_rule_id, so it is always
+        // present
+        // hence a check for null is not required
+        .when(simple("${body.messageRoutingRule.loginType} == 'OIDC'"))
+          .setProperty("credentials", simple("${body.messageRoutingRule.loginInformation}"))
+          .to("{{camel.route.oidc}}")
+        .otherwise()
+          .log("Unhandled loginType detected ${body.messageRoutingRule.loginType}")
+          .stop()
+      .end()
+      .to("{{camel.route.timestamp-notify}}");
+  }
+
+  void tokenOIDC() {
+    from("{{camel.route.oidc}}")
+      .routeId("oidc")
+      .onException(HttpOperationFailedException.class)
+        .continued(true)
+        .maximumRedeliveries("{{camel.max-redeliveries}}")
+        .redeliveryDelay("{{camel.redelivery-delay}}")
+        .useExponentialBackOff()
+        .backOffMultiplier(2)
+        .log(LoggingLevel.WARN, "Auth token request failed!")
+        .process(new DefaultTokenProcessor())
+        .marshal()
+        .json(JsonLibrary.Jackson)
+      .end()
+      .setHeader("CamelHttpMethod", constant("POST"))
+      .setHeader("Content-Type", simple("application/x-www-form-urlencoded"))
+      .setBody(
+          simple(
+              "grant_type=client_credentials&client_id=${exchangeProperty.credentials.clientID}&client_secret=${exchangeProperty.credentials.clientSecret}"))
+      .toD("${exchangeProperty.credentials.tokenURL}")
+      .log("auth response : ${body}")
+      .unmarshal()
+      .allowNullBody()
+      .json(JsonLibrary.Jackson)
+      .transform(simple("Bearer ${body[access_token]}"));
+  }
+
+  static class DefaultTokenProcessor implements Processor {
+
+    @Override
+    public void process(Exchange exchange) throws Exception {
+      HashMap<String, String> tokenMap = new HashMap<>();
+      tokenMap.put("access_token", "<token>");
+      exchange.getIn().setBody(tokenMap);
+    }
+  }
+
+  void timestampNotification() {
+    from("{{camel.route.timestamp-notify}}")
+      .routeId("timestamp-notify")
+      .onException(HttpOperationFailedException.class)
+        .handled(true)
+        .maximumRedeliveries("{{camel.max-redeliveries}}")
+        .redeliveryDelay("{{camel.redelivery-delay}}")
+        .useExponentialBackOff()
+        .backOffMultiplier(2)
+        .log("Timestamp request failed!")
+        .process(new DeadTimestampProcessor())
+        .to("jpa:org.dcsa.jit.persistence.entity.TimestampNotificationDead")
+      .end()
+      .setHeader("CamelHttpMethod", constant("POST"))
+      .setHeader("Content-Type", simple("application/json"))
+      .setHeader("Authorization", simple("${body}"))
+      .setBody(simple("${exchangeProperty.outboxMessage.payload}"))
+      .toD("${exchangeProperty.outboxMessage.messageRoutingRule.apiUrl}");
+  }
+
+  // inner class as its only required within TimestampRoutingService
+  static class DeadTimestampProcessor implements Processor {
```

Comment:
There is no need for an entire class, you can just refer to the method as .process(this::methodName)

This also applies to the other processor

Reply:
You're right, I did notice it while reviewing your PR too but I left it as a developer preference.